<launch>
    <arg name="use_sim_time" default="true"/>
    <arg name="log_level" default="warn"/>
    <arg name="rtk_enabled" default="true"/>
    

    <!-- ========================================= -->
    <!-- =============== STATIC TF =============== -->
    <!-- ========================================= -->
    <node name="base_link_tf" pkg="tf2_ros" exec="static_transform_publisher" args="0 0 0 0 0 0 /particle_filter /base_link"/>

    <!-- ========================================= -->
    <!-- =============== FIX TO POSE ============= -->
    <!-- ========================================= -->
    <node name="fix_to_pose" pkg="vector_map_visual_localizer" exec="fix_to_pose_node" output="log" args="--ros-args --log-level warn">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <remap from="pose" to="/sensing/gnss/ublox/pose"/>
        <remap from="fix_topic" to="/sensing/gnss/ublox/nav_sat_fix"/>
    </node>

    <!-- ========================================= -->
    <!-- ============== POSE TO POSE ============= -->
    <!-- ========================================= -->
    <node name="pose_to_path" pkg="vector_map_visual_localizer" exec="pose_to_path_node" output="log" args="--ros-args --log-level info">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="sub_topics" value="[/sensing/gnss/ublox/pose, /localization/pose_estimator/pose, /particle_pose]"/>
        <param name="pub_topics" value="[/sensing/gnss/ublox/path, /localization/pose_estimator/path, /particle_path]"/>
    </node>

    <!-- ========================================= -->
    <!-- ============== GROUND SERVER ============ -->
    <!-- ========================================= -->
    <node name="ground_server" pkg="vector_map_visual_localizer" exec="ground_server_node" output="screen" args="--ros-args --log-level info">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
    </node>

    <!-- ========================================= -->
    <!-- ============= TWIST ESTIMATOR =========== -->
    <!-- ========================================= -->
    <node name="twist_estimator" pkg="vector_map_visual_localizer" exec="twist_node" output="screen" args="--ros-args --log-level warn">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="rtk_enabled" value="$(var rtk_enabled)"/>
    </node>

    <!-- ========================================= -->
    <!-- ============ POSE INITIALIZER =========== -->
    <!-- ========================================= -->
    <node name="pose_initializer" pkg="vector_map_visual_localizer" exec="pose_init_node" output="log" args="--ros-args --log-level warn">
        <remap from="ublox_topic" to="/sensing/gnss/ublox/navpvt"/>
        <remap from="pose" to="/init/pose"/>
        <param name="rtk_enabled" value="$(var rtk_enabled)"/>
    </node>

</launch>
